runners/spark/src/main/java/org/apache/beam/runners/spark/metrics/MetricsAccumulator.java

```diff
@@ -58,13 +57,13 @@ public static void init(SparkPipelineOptions opts, JavaSparkContext jsc) {
               opts.isStreaming()
                   ? Optional.of(new CheckpointDir(opts.getCheckpointDir()))
                   : Optional.absent();
-          MetricsContainerStepMap metricsContainerStepMap = new MetricsContainerStepMap();
+          SparkMetricsContainerStepMap metricsContainerStepMap = new SparkMetricsContainerStepMap();
```

Comment:
Won't it be enough to just create new instance of `SparkMetricsContainerStepMap` here but keep the same interface of `MetricsContainerStepMap` everywhere as it was before get broken? Why do we need to change it everywhere?

Reply:
You're right, it's enough to change it just in initialization. I fixed it.

I will try to add a regression test.